rocAL/include/parameters/parameter_simple.h

```diff
@@ -35,11 +35,37 @@ class SimpleParameter : public Parameter<T> {
     T get() override {
         return _val;
     }
-    int update(T new_val) {
+
+    std::vector<T> get_array() override {
+        return _array;
+    }
+
+    void update_single_value(T new_val) {
         _val = new_val;
+    }
+
+    void update_array(T new_val) {
+        for (uint i = 0; i < _batch_size; i++) {
+            update_single_value(new_val);
+            _array[i] = _val;
+        }
+    }
+
+    int update(T new_val) {
+        if (_array.size())
+            update_array(new_val);
+        else
+            update_single_value(new_val);
         return 0;
     }
 
+    void create_array(unsigned batch_size) override {
+        if (_array.size() == 0)
+            _array.resize(batch_size);
+        _batch_size = batch_size;
+        update(_val);
```

Comment:
In other create_array calls update was not done, it is done here only
Any specific reason?

Reply:
Address this.
Also should you update the batch size and call update if _array.size != 0?